modules/packages/ArgumentParser.chpl

```diff
@@ -977,8 +977,11 @@ module ArgumentParser {
       _exitAfterHelp = exitAfterHelp;
 
       var _helpHandler = new shared HelpHandler();
+      // TODO: this fails if helpHandler is `borrowed`
       if !isNothingType(h) then
-        _helpHandler = helpHandler: shared h.chpl_t;
+        if isOwnedClass(helpHandler) || isUnmanagedClass(helpHandler)
+          then _helpHandler = shared.adopt(helpHandler);
+          else _helpHandler = helpHandler : shared;
```

Comment:
If we don't support borrowed here, what case is this covering? As far as I can tell it would just do nothing for a `shared`, and then error for a `borrowed`; if that's correct it might be better to just check if it's borrowed, and explicitly emit an error.

Reply:
The check is needed because there is no overload for adopting from a `shared`. I think it could be handled with 
```chapel
if !isSharedClass(helpHandler)
  then _helpHandler = shared.adopt(helpHandler);
  else _helpHandler = helpHandler;
```
This is definitely more clear, but still doesn't work for `borrowed`

Reply:
I see why the check is needed, but could we just as well get rid of the `else` branch here then?

Reply:
Oh wait, it has to set `_helpHandler` in the first place, not just cast `helpHandler`, nevermind!

Reply:
I think there is some clarity added by making the check `isSharedClass`, although the user facing error message is just as uninformative as before. So I added an explicit error for `borrowed` like you suggested.

Reply:
Sounds good to me. Just to double-check, the check would now be `!isSharedClass` (negated), right?

Reply:
Yes. It should be visible? GitHub is being weird and not syncing the new commit on the branch to this PR